plugins/destiny.py

```diff
@@ -312,6 +339,58 @@ def grim(text, bot):
     return output
 
 
+@hook.command('pvp')
+def pvp(text, bot):
+    if not text:
+        return pvp('help', bot)
+    text = text.split(" ")
+    if text[0].lower() == 'help':
+        return 'options: {}'.format(", ".join(PVP_OPTS))
+    membership = get_membership(text[0])
+    if type(membership) == str:
+        return membership
+    if len(text) == 1:  # if no stats are specified, add some
+        text.extend([
+            'k/d', 'bestSingleGameKills', 'longestKillSpree',
+            'longestSingleLife', 'orbsDropped', 'bestWeapon'])
+
+    output = []
+    for console in membership:
+        stats = get(
+            "{}Stats/Account/{}/{}/".format(
+                BASE_URL, console, membership[console]),
+            headers=HEADERS
+        ).json()['Response']['mergedAllCharacters']['results']['allPvP']['allTime']
+        tmp_out = []
+        for stat in text[1:]:
+            if stat in WEAPON_TYPES:
+                stat = "weaponKills{}".format(stat)
+            if stat in stats:
+                tmp_out.append('\x02{}\x02: {}'.format(
+                    stats[stat]['statId'], stats[stat]['basic']['displayValue']))
+            elif stat.lower() == 'k/d':
+                tmp_out.append('\x02k/d\x02: {}'.format(round(
+                    stats['kills']['basic']['value'] / stats['deaths']['basic']['value'], 2)))
+            elif stat == 'bestWeapon':
+                tmp_out.append('\x02bestWeapon\x02: {}'.format(best_weapon(stats)))
+            else:
+                tmp_out.append("Invalid option {}".format(stat))
+
+        output.append("{} - {}".format(CONSOLES[console - 1], ", ".join(tmp_out)))
+
+    return "; ".join(output)
+
+
+@hook.command('link')
+def link(text, nick, bot):
+    text = text.split(" ")
+    if text[0].lower == 'flush':
+        CACHE['links'][text[1]] = ""
+        return "{} flushed".format(text[1])
+    else:
+        CACHE['links'][nick] = text[0]
+        return "{} linked to {}".format(text[0], nick)
```

Comment:
this function is untested with the actual bot. from what i gathered from other bot functions, it will pass the nick of the user calling the function, if asked for it.